v2/protocol/http/protocol.go

```diff
@@ -201,7 +201,12 @@ func (p *Protocol) Respond(ctx context.Context) (binding.Message, protocol.Respo
 		if !ok {
 			return nil, nil, io.EOF
 		}
+
+		if in.msg == nil {
+			return nil, in.respFn, in.err
```

Comment:
ah, we got caught by the typed nil in go... nice. 👍

Reply:
Hehe looks like it, a very vicious one!